docs/proposals/config-management-plugin-v2.md

```diff
@@ -0,0 +1,304 @@
+with argocd-cmp-server as it’s entrypoint:
+
+```bash
+# A plugin is a container image which runs as a sidecar, with the execution environment
+# necessary to render manifests. To install a plugin, 
+containers:
+- name: cdk8s
+  command: [/var/run/argocd/argocd-cmp-server]
+  image: cdk8s/cdk8s:latest
+  volumeMounts:
+  - mountPath: /var/run/argocd
+    name: var-files
+```
+
+The argocd-cmp-server binary will be populated inside the plugin container via an init container in the argocd-repo-server, 
+which will pre-populate a volume shared between plugins and the repo-server.
+
+```bash
+# An init container will copy the argocd static binary into the shared volume
+# so that the CMP server can become the entrypoint
+initContainers:
+- command:
+  - cp
+  - -n
+  - /usr/local/bin/argocd
+  - /var/run/argocd/argocd-cmp-server
+  image: quay.io/argoproj/argocd:latest
+  name: copyutil
+  volumeMounts:
+  - mountPath: /var/run/argocd
+    name: var-files
+ 
+# var-files is a shared volume between repo-server and cmp-server which holds:
+# 1) socket files that repo-server uses to communicate to each plugin
+# 2) git repositories cloned by repo-server
+volumes:
+- emptyDir: {}
+  name: var-files
+```
+
+#### Configuration
+
+Plugins will be configured via a ConfigManagementPlugin manifest located inside the plugin container, placed at a 
+well-known location (e.g. /plugin.yaml). Argo CD is agnostic to the mechanism of how the plugin.yaml would be placed, 
+but various options can be used on how to place this file, including: 
+- Baking the file into the plugin image as part of docker build
+- Volume mapping the file through a configmap.
+
+Note that, while the ConfigManagementPlugin looks like a Kubernetes object, it is not actually a custom resource. 
+It only follows kubernetes-style spec conventions.
+
+```bash
+# metadata file is in the root and shell executor knows about it
+apiVersion: argoproj.io/v1alpha1
+kind: ConfigManagementPlugin
+metadata:
+  name: cdk8s
+spec:
+  version: v1.0
+  init:
+    command: [cdk8s, init]
+  generate:
+    command: [sh, -c, "cdk8s synth && cat dist/*.yaml"]
+  discovery:
+    find:
+    - command: [find . -name main.ts]
+      glob: "**/*/main.ts"
+    check:
+    - command: [-f ./main.ts]
+      glob: "main.ts"
+  allowConcurrency: true
+```
+
+#### Config Management Plugin API Server (cmp-server)
+The Config Management Plugin API Server (cmp-server) will be a new Argo CD component whose sole responsibility will be 
+to execute `generate` commands inside the plugin environment (the sidecar container), at the request of the repo-server.
+
+The cmp-server will expose the following APIs to the repo-server,
+
+- GenerateManifests(path) - returns YAML output using plugin tooling
+- IsSupported(path) - returns whether or not the given path is supported by the plugin
+
+At startup, cmp-server looks at the /plugin.yaml ConfigManagementPlugin specification file to understand how to perform the requests.
```

Comment:
Let's avoid a root file system path. We could use `/home/argocd-cmp/plugin.yaml`